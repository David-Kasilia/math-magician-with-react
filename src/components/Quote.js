import React from 'react';

const Quote = () => (
  <p className="quote">
    Mathematics is about numbers, equations,computations, or algorithms: its is
    about understanding.- William Paul Thurston
  </p>
);

export default Quote;
