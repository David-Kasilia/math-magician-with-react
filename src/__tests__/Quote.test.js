import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import Quote from '../components/Quote';

test('Unit Test Check For Quote Component', () => {
  render(<Quote />);
  const linkElement = screen.getByText(/Mathematics is about numbers/i);
  expect(linkElement).toBeInTheDocument();
});

describe('Unit Test For Quote component', () => {
  it('Quote Component renders correctly', () => {
    const tree = TestRenderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
