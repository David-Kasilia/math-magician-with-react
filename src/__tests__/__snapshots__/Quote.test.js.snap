// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test For Quote component Quote Component renders correctly 1`] = `
<p
  className="quote"
>
  Mathematics is about numbers, equations,computations, or algorithms: its is about understanding.- William Paul Thurston
</p>
`;
